src/exo/LoopIR_compiler.py

```diff
@@ -397,10 +397,10 @@ def from_lines(x):
         else:
             is_public_decl = id(p) in orig_procs
 
+            p = ParallelAnalysis().run(p)
```

Comment:
Why?

Reply:
As in why is it first? Or why at all?

Reply:
Why did you move it to the first?

Reply:
I don't think the current effects analysis handle frees correctly, so I moved it first. We'll probably have to revisit this ordering later for GPUs, but in the current state, I don't think the order particularly matters?

Reply:
Okay yeah that's fine